src/components/container/GarminMonthCharts/GarminMonthCharts.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react'
+import { DateRangeNavigator } from '../../presentational';
+import DeviceDataMonthChart from '../DeviceDataMonthChart'
+import language from '../../../helpers/language';
+import { DailyDataType } from '../../../helpers/query-daily-data';
+
+export interface GarminMonthChartsProps {
+	previewState?: GarminMonthChartsPreviewState
+}
+
+export type GarminMonthChartsPreviewState = "notEnabled" | "notConnected" | "connected";
+
+export default function (props: GarminMonthChartsProps) {
+	const [visible, setVisible] = useState(false);
+	var currentDate = new Date();
+	const [intervalStart, setIntervalStart] = useState(new Date(currentDate.getFullYear(), currentDate.getMonth(), 1, 0, 0, 0, 0));
+
+	function updateInterval(date: Date) {
+		setIntervalStart(date);
+	}
+
+	function onDataDetected() {
+		setVisible(true);
+	}
+
+	var month = intervalStart.getMonth();
+	var year = intervalStart.getFullYear();
+	return (
+		<div style={{ display: visible ? "block" : "none" }}>
+			<DateRangeNavigator intervalType="Month" intervalStart={intervalStart} onIntervalChange={updateInterval} />
+			<DeviceDataMonthChart
+				onDataDetected={() => onDataDetected()}
+				previewState={props.previewState == "connected" ? "WithData" : props.previewState == "notEnabled" || props.previewState == "notConnected" ? "NoData" : undefined }
+				lines={[{
+					showAverage: true,
+					dailyDataType: DailyDataType.GarminSteps,
```

Comment:
This again looks like the only thing that differs between this and the Fitbit version?  This should be a configurable option on a shared widget.

Reply:
I feel like this might be a little different than the other components because each device type might have different data types to display. I only implemented Steps and HR here which do happen to both be implemented in each device, but maybe there are data types that are only in one or the other (Garmin Body Battery for example). 

With that said maybe there is a way to structure this that would support this. thoughts @ajmenca?

Reply:
Side note: I guess the `previewState` needs to also be defined on the wrapper and passed down to `MonthCharts` too?  But that's a quirk of our storybook setup mostly.

Reply:
Also, with all this said... I pretty much have this working locally in this form to ensure that I wasn't giving you a bad recommendation 😆 

I can just push it, if you'd like.

Reply:
Ha.. that is what I was thinking as I was reading through these.

In the immortal words of Salt N Pepa I would say "push it"